### PR TITLE
Merge MintMaker and MPP-Renovate images

### PR DESCRIPTION
**Pull request by FernandesMF**
We have been maintaining two different images, one for our runners in MPP, and another for Konflux. This commit merges the two images, which can now be used in both environments and will reduce our maintenance efforts (hopefully).

The most important changes with respect to the MPP image are:
- this image is based on ubi9-minimal (the MPP image was based on fedora)
- this image lacks python 3.8/3.10, as ubi9 does not have packages for these versions
- this image does not correction to rh-ca bundle mount

The moist important changes with respect to the previous version are:
- drop installation of `npm` and `nodejs` packages via the package manager
- manual installation of `nodejs`

There will be a follow-up change in the MPP deployment files. This change must be merged first.

JIRA: [CWFHEALTH-3276]

### Testing remarks
I made dry runs in the both MPP and Konflux stage environments. The image worked fine in both.

A few minor remarks:
  - MPP: there was a lookup error with a go package (kubevirt.io/api); this seems to be unrelated to the image itself
  - STN: there was an authentication error due to revoked token, but this error was already present before the current image changes; since the image worked fine in MPP for the same authentication, this is not related tothe image itself

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,11 +12,14 @@ LABEL description="Mintmaker - Renovate custom image" \
...
+
+ARG NODEJS_VERSION=20.17.0
 
 # Using OpenSSL store allows for external modifications of the store. It is needed for the internal Red Hat cert.
 ENV NODE_OPTIONS=--use-openssl-ca
 
+
 RUN microdnf update -y && \
     microdnf module enable -y nodejs:20/common && \
     microdnf install -y \
```
Comment (staticf0x) [inline Dockerfile:25]:
```suggestion
RUN microdnf update -y && \
    microdnf install -y \
```

File: Dockerfile [hunk 2]
```diff
@@ -12,11 +12,14 @@ LABEL description="Mintmaker - Renovate custom image" \
       url="https://github.com/konflux-ci/mintmaker-renovate-image/" \
       vendor="Red Hat, Inc."
 
-ARG RENOVATE_VERSION=37.413.2-custom
+ARG RENOVATE_VERSION=38.55.2-rpm
+
+ARG NODEJS_VERSION=20.17.0
```
Comment (staticf0x) [inline Dockerfile:17]:
```suggestion
ARG RENOVATE_VERSION=38.55.2-rpm

# NodeJS version used for Renovate, has to satisfy the version
# specified in Renovate's package.json
ARG NODEJS_VERSION=20.17.0
```

I'd just put this comment here to warn against just updating this randomly :smile:

File: Dockerfile [hunk 3]
```diff
@@ -27,13 +30,25 @@ RUN microdnf update -y && \
         python3.11-pip \
         python3-pip \
         python3-dnf \
-        nodejs \
-        npm \
-        skopeo && \
+        python3.9 \
+        cargo \
+        golang \
+        podman \
```
Comment (staticf0x) [inline Dockerfile:36]:
```suggestion
```

I think we can drop podman, it was required before skopeo became the go-to thing.

**Review by staticf0x — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -27,13 +30,25 @@ RUN microdnf update -y && \
...
         https://github.com/tektoncd/cli/releases/download/v0.35.1/tektoncd-cli-0.35.1_Linux-64bit.rpm
 
+# Install nodejs
+RUN curl -o node-v${NODEJS_VERSION}-linux-x64.tar.xz https://nodejs.org/dist/v${NODEJS_VERSION}/node-v${NODEJS_VERSION}-linux-x64.tar.xz
+RUN tar xf node-v${NODEJS_VERSION}-linux-x64.tar.xz && \
+    mv node-v${NODEJS_VERSION}-linux-x64/bin/* /bin/ && \
+    mv node-v${NODEJS_VERSION}-linux-x64/include/* /include/ && \
+    mv node-v${NODEJS_VERSION}-linux-x64/lib/* /lib/ && \
+    rm -r node-v${NODEJS_VERSION}-linux-x64 && \
+    rm -r node-v${NODEJS_VERSION}-linux-x64.tar.xz
```
Comment (qixiang) [inline Dockerfile:50]:
```suggestion
    rm -fr node-v${NODEJS_VERSION}-linux-x64 && \
    rm -f node-v${NODEJS_VERSION}-linux-x64.tar.xz
```

File: Dockerfile [hunk 5]
```diff
@@ -12,11 +12,14 @@ LABEL description="Mintmaker - Renovate custom image" \
...
 
-ARG RENOVATE_VERSION=37.413.2-custom
+ARG RENOVATE_VERSION=38.55.2-rpm
+
+ARG NODEJS_VERSION=20.17.0
 
 # Using OpenSSL store allows for external modifications of the store. It is needed for the internal Red Hat cert.
 ENV NODE_OPTIONS=--use-openssl-ca
 
+
```
Comment (qixiang) [inline Dockerfile:22]:
```suggestion
```

**Comment by FernandesMF** [comment]
Hey team, thank you for the reviews! I applied the suggestions and squashed the changes in a single commit.
Could you have a second look?

**Review by staticf0x — APPROVED** [review]
LGTM

**Pull request merged**